### PR TITLE
[MISC] Refactor and address most errcheck linter ignores

### PR DESCRIPTION
**Pull request by nightah**
This is mostly a quality of life change.
When we first implemented the errcheck linter we ignored a number of items in our legacy codebase with intent to revisit down the track.

There's a only a few leftover TODO's, these have been left for the team to discuss on Matrix and agree on the way forward.

### CODE REVIEW

**Review by james-d-elliott — APPROVED** [review]
LGTM

File: internal/handlers/handler_firstfactor.go [hunk 1]
```diff
@@ -95,7 +95,8 @@ func FirstFactorPost(msInitialDelay time.Duration, delayEnabled bool) middleware
 
 		if err != nil {
 			ctx.Logger.Debugf("Mark authentication attempt made by user %s", bodyJSON.Username)
-			ctx.Providers.Regulator.Mark(bodyJSON.Username, false) //nolint:errcheck // TODO: Legacy code, consider refactoring time permitting.
+
+			_ = ctx.Providers.Regulator.Mark(bodyJSON.Username, false)
```
Comment (clems4ever) [inline internal/handlers/handler_firstfactor.go:99]: Why not handle the error here?

File: internal/handlers/handler_firstfactor.go [hunk 2]
```diff
@@ -104,7 +105,8 @@ func FirstFactorPost(msInitialDelay time.Duration, delayEnabled bool) middleware
 
 		if !userPasswordOk {
 			ctx.Logger.Debugf("Mark authentication attempt made by user %s", bodyJSON.Username)
-			ctx.Providers.Regulator.Mark(bodyJSON.Username, false) //nolint:errcheck // TODO: Legacy code, consider refactoring time permitting.
+
+			_ = ctx.Providers.Regulator.Mark(bodyJSON.Username, false)
```
Comment (clems4ever) [inline internal/handlers/handler_firstfactor.go:109]: why not handle the error here?

File: web/tsconfig.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react",
+    "jsx": "react-jsx",
```
Comment (clems4ever) [inline web/tsconfig.json:20]: ah yeah, I've seen that one at work XD.

**Review by clems4ever — APPROVED** [review]
Only a few comments but I'm ok for merging it like this since it already improves the situation a lot.

File: internal/handlers/handler_firstfactor.go (hunk 1 above)
Comment (nightah) [reply internal/handlers/handler_firstfactor.go:99]: I had issues when running integration tests and I thought it was related.

After some testing it appears that it's not so I'll handle these too.

Comment (nightah) [reply internal/handlers/handler_firstfactor.go]: Addressed in https://github.com/authelia/authelia/pull/1511/commits/15646c4bedd408da2fe42ddef2c07318c3a54c25.

File: internal/handlers/handler_firstfactor.go (hunk 2 above)
Comment (nightah) [reply internal/handlers/handler_firstfactor.go]: Addressed in https://github.com/authelia/authelia/pull/1511/commits/15646c4bedd408da2fe42ddef2c07318c3a54c25.

**Pull request merged**